dags/mag_and_friends.py

```diff
@@ -426,22 +435,30 @@ def update_provenance():
                                     # Write back to the file
                                     writer.writerows(all_data.values())
 
+                            update_provenance = PythonOperator(
+                                task_id="update_provenance",
+                                python_callable=update_provenance_callable,
+                                task_group=current_tasks
+                            )
+
                             copy_study_to_cluster >> \
                             run_mag >> \
                             watch_mag >> \
-                            copy_results_from_cluster >> \
-                            post_process_results() >> \
-                            update_provenance()
+                            update_provenance
 
                             copy_study_to_cluster >> \
                             run_metatdenovo >> \
                             watch_metatdenovo >> \
-                            copy_results_from_cluster 
+                            copy_metatdenovo_results_from_cluster >> \
+                            post_process_results >> \
+                            update_provenance
                             
                             copy_study_to_cluster >> \
                             run_taxprofiler >> \
                             watch_taxprofiler >> \
-                            copy_results_from_cluster
+                            copy_taxprofiler_results_from_cluster >> \
+                            post_process_results >> \
+                            update_provenance
 
                             if os.path.exists(accessionsFile):
```

Comment:
I see it is done via a manual loop. I assume the dynamic task mapping is still unusable for complex tasks like these?

Reply:
i spent a bit of time, maybe a day or two, trying to sort it out w dynamic task mapping. i decided a few things:
1. even if i got it to work itd take me enough time it likely wasnt worth attempting for an intial implementation
2. it was producing code that was harder to read (and consequently maintain long term) 
3. it seemed somewhat brittle. it didnt work unless you did it *exactly* right. even things that intuitively seemed they should work didnt

its a shame, cause id have preferred to do it the 'airflow-y' way.. but not something i was so attached to as to give myself a headache over it.